File: hw/vargas/u3/hw29Singleton/python/ec/edu/espe/singleton/model/date_model.py
class DateModel:

    def __init__(self, day: int = 0, month: int = 0, year: int = 0, hour: int = 0, minute: int = 0):
        self.day = day
        self.month = month
        self.year = year
        self.hour = hour
        self.minute = minute

    def __str__(self):
        return f"{self.year:04d}-{self.month:02d}-{self.day:02d} @ {self.hour:02d}:{self.minute:02d}"

    def is_before(self, other: 'DateModel') -> bool:
        if self.year != other.year:
            return self.year < other.year
        if self.month != other.month:
            return self.month < other.month
        if self.day != other.day:
            return self.day < other.day
        if self.hour != other.hour:
            return self.hour < other.hour
        return self.minute < other.minute

File: hw/vargas/u3/hw29Singleton/javaScript/ec/edu/espe/singleton/model/DateModel.js

export class DateModel {
    constructor(day = 0, month = 0, year = 0, hour = 0, minute = 0) {
        this.day = day;
        this.month = month;
        this.year = year;
        this.hour = hour;
        this.minute = minute;
    }

    toString() {
        const pad = (num) => String(num).padStart(2, '0');
        const yearStr = String(this.year).padStart(4, '0');
        return `${yearStr}-${pad(this.month)}-${pad(this.day)} @ ${pad(this.hour)}:${pad(this.minute)}`;
    }

    isBefore(other) {
        if (this.year !== other.year) return this.year < other.year;
        if (this.month !== other.month) return this.month < other.month;
        if (this.day !== other.day) return this.day < other.day;
        if (this.hour !== other.hour) return this.hour < other.hour;
        return this.minute < other.minute;
    }
}